recipes/cookbook_uploader.rb

```diff
@@ -0,0 +1,177 @@
+#
+# Cookbook Name:: osl-jenkins
+# Recipe:: cookbook-uploader
+#
+# Copyright (C) 2015 Oregon State University
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# git and github: for git
+# build-token-root: for triggering from Github PRs
+# parameterized-trigger: for triggering other jobs (e.g. the environment-bumper
+#   job) with parameters
+# text-finder: for marking unstable builds (used in this case to mark builds
+#   where no action was taken)
+%w(git github build-token-root parameterized-trigger text-finder).each do |p|
+  jenkins_plugin p do
+    notifies :restart, 'service[jenkins]'
+  end
+end
+
+org_name = node['osl-jenkins']['cookbook_uploader']['org']
+chef_repo = node['osl-jenkins']['cookbook_uploader']['chef_repo']
+
+# A build is triggered on every Github comment, but will only succeed if the
+# comment was a bump request.  This message is displayed if the comment was not
+# a bump request and the build will be marked as unstable.
+non_bump_message = 'Exiting because comment was not a bump request'.freeze
+
+# Note: The `github_user` field is used for Git pull/push over https in
+# conjuction with the token, rather than an SSH key.
+begin
+  secrets = Chef::EncryptedDataBagItem.load(
+    node['osl-jenkins']['cookbook_uploader']['databag'],
+    node['osl-jenkins']['cookbook_uploader']['secrets_item']
+  )
+rescue
```

Comment:
We should only be rescuing if the chef server returns a 404 meaning the data bag does not exist. Everything else we should still raise and exception like how it is here: https://github.com/osuosl-cookbooks/osl-rsync/blob/master/recipes/default.rb#L27
Otherwise if someone pushed a bag data bag config it would silently reset the data to an empty array and we would have no way of knowing that happened.